src/uct/tcp/tcp_cm.c

```diff
@@ -293,12 +296,20 @@ uct_tcp_ep_t *uct_tcp_cm_get_ep(uct_tcp_iface_t *iface,
     }
 
     ep = ucs_container_of(elem, uct_tcp_ep_t, elem);
-    ucs_assert(ep->flags & (with_ctx_cap | UCT_TCP_EP_FLAG_ON_MATCH_CTX));
+    ucs_assert(ep->flags & UCT_TCP_EP_FLAG_ON_MATCH_CTX);
+
+    if ((queue_type == UCS_CONN_MATCH_QUEUE_UNEXP) ||
+        !(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_TX)) {
+        ucs_assert(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_RX);
+    }
 
     if (remove_from_ctx) {
         ucs_assert((ep->flags & UCT_TCP_EP_CTX_CAPS) ==
                    UCT_TCP_EP_FLAG_CTX_TYPE_RX);
         ep->flags &= ~UCT_TCP_EP_FLAG_ON_MATCH_CTX;
+        /* The EP was removed from connection matching, move it to the EP list
+         * on iface to be able to destroy it from iface cleanup */
```

Comment:
pls specify which function does list_del() in this case

Reply:
done